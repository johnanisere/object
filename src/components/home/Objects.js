import React from "react"
import PropTypes from "prop-types"
import { MdRemoveRedEye } from "react-icons/md";
import OptionLayout from "../SectionLayout"
import List from "../../blocks/FlexibleContainer"

const App = ({ displayData }) => {
    const title = <span><MdRemoveRedEye color="#8e8e93" /> {`  OBJECTS [${displayData.length}]`}</span>

    return (
        <OptionLayout title={title}>
            <List list="true" direction="column">
                {
                    displayData.map(({ number }, key) => <div key={key} className="list"><MdRemoveRedEye color="#8e8e93" />  OBJ {number}</div>)
                }
            </List>
        </OptionLayout>
    )
}
App.propTypes = {
    displayData: PropTypes.array.isRequired,
}
export default App