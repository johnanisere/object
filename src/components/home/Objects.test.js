
import React from "react";
import { shallow } from "enzyme";

import Objects from "./Objects"

import OptionLayout from "../SectionLayout"
import List from "../../blocks/FlexibleContainer"

const props = {
    displayData: [],
}

it("Objects component exists", () => {
    const wrapper = shallow(<Objects {...props} />);
    expect(wrapper.exists()).toBe(true);
});

it("Objects component renders", () => {
    const wrapper = shallow(<Objects {...props} />);
    expect(wrapper).toMatchSnapshot();
});

it(`Objects should contain "OptionLayout",  "List", `, () => {
    const wrapper = shallow(<Objects {...props} />);

    expect(wrapper.find(OptionLayout)).toHaveLength(1);
    expect(wrapper.find(List)).toHaveLength(1);
});