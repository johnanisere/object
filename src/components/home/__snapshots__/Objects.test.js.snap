// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Objects component renders 1`] = `ShallowWrapper {}`;
